### ISSUE TITLE
[Enh] Add [external] tunneling support

### ISSUE THREAD

**Issue by AnrDaemon** [xdebug/vscode-php-debug#286]
Even just a command to launch external tunnel program to receive debugging connection would simplify setup.
Right now, I launch SSH tunnel manually to open the session for debugging.

**Comment by btry** [comment]
Hi

I agree, It would be very convenient to use ssh port forwarding features to easily debug a remote server

**Comment by marabesi** [comment]
up

**Comment by zobo** [comment]
I feel this should be responsibility of a different VS Code extension. You could also use `tasks.json`, I'm sure there is a way.
Also, if possible, I strongly suggest VS Code Remote SSH.

**Comment by AnrDaemon** [comment]
I tried remote SSH, it is inconvenient on intermittent connection.
Yes, I can configure shell profile as SSH with tunneling. Guess I have to…

**Comment by zobo** [comment]
Yes, if the connection is unstable then it can probably be a pain.
If you don't have a defined workflow, you can probably make a task like here https://code.visualstudio.com/docs/editor/tasks and execute ssh with the propper -L or -R parameters.

I think you could also couple this with `preLaunchTask` in `launch.json`. https://code.visualstudio.com/docs/editor/debugging#_launchjson-attributes

Good luck!

**Comment by AnrDaemon** [comment]
Yes, that's a good idea.
Worth adding to the documentation, don't you think?

**Comment by zobo** [comment]
I'll leave it open, and figure out where to document cases like this... I think it's a bit too much for the main README, but I should definitely make a Wiki or a "product page".
